src/coreclr/tests/src/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -113,189 +114,173 @@ public static unsafe bool ListChildPids(int ppid, out int[] buffer)
-                    string[] childrenPidAsStrings = File.ReadAllText(childrenFile).Split(' ', StringSplitOptions.RemoveEmptyEntries);
-                    foreach (var childPidAsString in childrenPidAsStrings)
+                    try
                     {
-                        int childPid = int.Parse(childPidAsString);
-                        Process childProcess = Process.GetProcessById(childPid);
-                        if (childProcess.ProcessName.Equals(childName, StringComparison.Ordinal))
-                        {
-                            child = childProcess;
-                            return true;
-                        }
-                        else
-                        {
-                            childrenFilesToCheck.Enqueue($"/proc/{childPid}/task/{childPid}/children");
-                        }
+                        children.Add(Process.GetProcessById(pid));
+                    }
+                    catch (ArgumentException)
+                    {
+                        // Ignore failure to get process, the process may have exited
                     }
                 }
-                catch (IOException)
+            }
+            catch (IOException)
+            {
+                // Ignore failure to read process children data, the process may have exited
+            }
+
+            return children;
+        }
+
+        private static IEnumerable<Process> MacOS_GetChildren(Process process)
+        {
+            var children = new List<Process>();
+            if (libproc.ListChildPids(process.Id, out int[] childPids))
+            {
+                foreach (var childPid in childPids)
                 {
-                    // Ignore failure to read process children data, the process may have exited.
+                    children.Add(Process.GetProcessById(childPid));
                 }
             }
 
-            child = null;
+            return children;
+        }
+    }
+
+    public class CoreclrTestWrapperLib
+    {
+        public const int EXIT_SUCCESS_CODE = 0;
+        public const string TIMEOUT_ENVIRONMENT_VAR = "__TestTimeout";
+        
+        // Default timeout set to 10 minutes
+        public const int DEFAULT_TIMEOUT = 1000 * 60*10;
+
+        public const string COLLECT_DUMPS_ENVIRONMENT_VAR = "__CollectDumps";
+        public const string CRASH_DUMP_FOLDER_ENVIRONMENT_VAR = "__CrashDumpFolder";
+
+        static bool CollectCrashDump(Process process, string path)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                using (var crashDump = File.OpenWrite(path))
+                {
+                    var flags = DbgHelp.MiniDumpType.MiniDumpWithFullMemory | DbgHelp.MiniDumpType.MiniDumpIgnoreInaccessibleMemory;
+                    return DbgHelp.MiniDumpWriteDump(process.Handle, process.Id, crashDump.SafeFileHandle, flags, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+                }
+            }
```

Comment:
Since you're already rewriting this completely, it may be worth trying to switch to `createdump.exe` so we're consistent across platforms (and get more usage of `createdump.exe`).